### PR TITLE
Match Chrome User Agent in M-SEARCH SSDP packets

### PR DESCRIPTION
**Pull request by ShivanKaul**
Resolves https://github.com/brave/brave-browser/issues/18017

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

Same as https://github.com/brave/brave-browser/issues/18017#issuecomment-1924661227

### CODE REVIEW

File: chromium_src/chrome/browser/media/router/discovery/dial/dial_service_impl.cc [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+#include "components/version_info/version_info.h"
+
+namespace version_info {
+constexpr std::string_view GetProductNameForChrome() {
+  return "Google Chrome";
+}
+}  // namespace version_info
+
+#define GetProductName GetProductNameForChrome
```
Comment (goodov) [inline chromium_src/chrome/browser/media/router/discovery/dial/dial_service_impl.cc:16]: this only changes the product name. The user agent will look pretty weird for the receiving side.

Comment (ShivanKaul) [reply chromium_src/chrome/browser/media/router/discovery/dial/dial_service_impl.cc:16]: What do you mean? The User Agent with this change looks like: `USER-AGENT: Google Chrome/122.1.64.0 Mac OS X\r\n` on Mac. Do you mean the version string is out of sync?

Comment (goodov) [reply chromium_src/chrome/browser/media/router/discovery/dial/dial_service_impl.cc:16]: `Google Chrome/122.1.64.0`

`Google Chrome` version has different values in minor/build/patch fields, i.e. minor is always 0, build is always in thousands. Not sure how critical this is in terms of compatibility, that's why I added @fmarier to have a second look here.

Comment (ShivanKaul) [reply chromium_src/chrome/browser/media/router/discovery/dial/dial_service_impl.cc:18]: I changed it so as to match the Version Number too.

Comment (fmarier) [reply chromium_src/chrome/browser/media/router/discovery/dial/dial_service_impl.cc:18]: @ShivanKaul says the new version number looks like this now:
```
Google Chrome/122.0.6261.29 Mac OS X\r\n
```

**Review by fmarier — APPROVED** [review]

**Review by goodov — APPROVED** [review]

**Pull request merged**